### ISSUE TITLE
upgrade fabric8

### ISSUE THREAD

**Issue by rawlingsj** [fabric8io/fabric8#4655]
It would be good to provide a command to upgrade the fabric8 components to a new released version.  It might need a bit of thinking so that rolling updates are used and support for using different strategies like big bang as opposed to progressively by namespace.  

A rollback feature would need to be provided and possibly cater for the update script being cancelled / interrupted?

**Comment by jstrachan** [comment]
agreed - we should be able to replace the templates and use the kubernetes rolling upgrade hopefully?

I don't see any way to roll forward/backward templates mind you; without storing them in a git repo and using git etc

**Comment by Morriz** [comment]
Any updates on this? Or a link to the manual steps to do this? I installed with the fabric8-installer, and let it do it's magic. I am used to pulling repos and reloading vagrant and such, but find this too obscure in the fabric8 context.
